File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int csum = 0;
        int msum = nums[0];
        for (int i: nums) {
            csum += i;
            msum = max(msum, csum);
            csum = max(csum, 0);
        }
        return msum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans = nums[0]
        cur_sum = nums[0]
        for i in nums[1:]:
            cur_sum = max(i, cur_sum + i)
            ans = max(ans, cur_sum)
        return ans